File: API/src/main/java/com/studentbox/api/service/forum/PostService.java
package com.studentbox.api.service.forum;

import com.studentbox.api.entities.forum.Post;
import com.studentbox.api.models.common.PageModel;
import com.studentbox.api.models.common.PaginationModel;
import com.studentbox.api.models.post.PostCreationModel;
import com.studentbox.api.models.post.PostModel;
import com.studentbox.api.models.post.reply.PostReplyCreationModel;

import java.util.List;

public interface PostService {
    PageModel<PostModel> getPage(PaginationModel paginationModel);
    Post findById(String id);
    PostModel findBasicById(String id);
    void create(PostCreationModel postCreationModel);
    void delete(String id);
    void update(String id, PostCreationModel postModel);
    void like(String id);
    void createReply(String postId, PostReplyCreationModel postReplyModel);
    void updateReply(String postId, String replyId, PostReplyCreationModel postReplyModel);
    void deleteReply(String postId, String replyId);
    void likeReply(String postId, String replyId);
}

File: frontend/studentbox/src/service/postService.js
import axios from "../custom-axios/axios";

const postService = {

    fetchAllPosts: () => axios.get("/posts"),

    fetchPost: (postId) => axios.get(`/posts/${postId}`),

    addPost: (postCreationModel) => axios.post("/posts",{
       "postCreationModel":postCreationModel
    }),

    updatePost: (postId,postCreationModel) => axios.patch(`/posts/${postId}`,{
       "postCreationModel":postCreationModel
        }
    ),

    deletePost: (postId) => axios.delete(`/posts/${postId}`),

    likePost: (postId) => axios.put(`/posts/${postId}/like`),

    addReply : (postId, postReplyCreationModel) => axios.post(`/posts/${postId}/reply`,{
        "postReplyCreationModel":postReplyCreationModel
    }),

    updateReply: (postId, replyId, postReplyCreationModel) => axios.patch(`/posts/${postId}/reply/${replyId}/`,{
        "postReplyCreationModel":postReplyCreationModel
    }),

    deleteReply: (postId, replyId) => axios.delete(`/posts/${postId}/reply/${replyId}`),

    likeReply: (postId, replyId) => axios.patch(`/posts/${postId}/reply/${replyId}/like`)
}

export default postService
